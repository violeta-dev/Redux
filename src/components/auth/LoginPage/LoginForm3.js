import React from 'react';
import T from 'prop-types';
import { Button, Checkbox, Input } from 'antd';
import { MailOutlined, LockOutlined } from '@ant-design/icons';

import useForm from '../../../hooks/useForm';
import styles from './LoginForm.module.css';

function LoginForm({ onSubmit }) {
  const [form, handleChange] = useForm({
    email: '',
    password: '',
    remember: false,
  });
  const { email, password, remember } = form;

  const canSubmit = () => {
    return !!(email && password);
  };

  const handleSubmit = ev => {
    ev.preventDefault();
    onSubmit(form);               
  };

  return (   
    <form onSubmit={handleSubmit}>
      <Input
        name="email"
        className={styles.input}
        prefix={<MailOutlined />}
        placeholder="Email"
        onChange={handleChange}
        value={email}
      />
      <Input.Password
        name="password"
        className={styles.input}
        prefix={<LockOutlined />}
        placeholder="Password"
        onChange={handleChange}
        value={password}
      />
      <Checkbox
        name="remember"
        className={styles.input}
        onChange={handleChange}
        checked={remember}
      >
        Remember me
      </Checkbox>
      <Button type="primary" htmlType="submit" disabled={!canSubmit()} block>
        Log In
      </Button>
    </form>
  );
}

LoginForm.propTypes = {
  onSubmit: T.func.isRequired,
};

export default LoginForm;
