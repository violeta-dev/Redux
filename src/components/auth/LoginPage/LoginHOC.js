import React, { Component } from 'react';

import { connect } from 'react-redux';




const LoginHOC = (PassedComponent) => {
  class LoginParent extends Component {
    

    constructor() {
      super();

      this.state = {
        email: '',
        password: '',
        remember: false
      };
      
      this.handleChange = this.handleChange.bind(this);
      this.handleSubmit = this.handleSubmit.bind(this);
      this.canSubmit = this.canSubmit.bind(this);
    }

    

    handleChange(e) {
      const { name, value } = e.target;
      this.setState({
        [name]: value,
      });
    }
   
    canSubmit = () => {
      const { email, password } = this.state;
      return !!(email && password);
    };

    handleSubmit(e) {
      e.preventDefault();
      const { email, password, remember } = this.state;
      const data = {
        email,
        password,
        remember
      };

      this.props.onSubmit(data);
    }


    render() {
      const { email, password, remember } = this.state;
      return (
        <PassedComponent
          email={email}
          password={password}
          remember= {remember}
          handleChange={this.handleChange}
          handleSubmit={this.handleSubmit}
          canSubmit={this.canSubmit()}
        />
      )
    }
  }


function mapDispatchToProps(state){
 return state

};

  return connect(mapDispatchToProps)(LoginParent)
}

export default LoginHOC;