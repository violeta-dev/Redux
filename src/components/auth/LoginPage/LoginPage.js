import React from 'react';
import T from 'prop-types';
import { Alert, Col, Row, Typography } from 'antd';
//import { connect } from 'react-redux';


import { login } from '../../../api/auth';
import LoginForm from './LoginForm';



const { Title } = Typography;

export class LoginPage extends React.Component {
  state = {
    error: null,
  };

  handleSubmit = async (credentials) => {
    const { onLogin,location, history } = this.props;
    this.resetError();
    try {
      const isLogged= await login(credentials)
      const { from } = location.state || { from: { pathname: '/' } };    
      onLogin(isLogged);
      history.replace(from);     
    } catch (error){
      this.setState({ error });
    }
  };

  resetError = () => this.setState({ error: null });

  render() {
    const { error } = this.state;
    return (
      <Row>
        <Col span={8} offset={8} style={{ marginTop: 64 }}>
          <Title style={{ textAlign: 'center' }}>Log In</Title>
          <LoginForm onSubmit={this.handleSubmit} />
          {error && (
            <Alert
              afterClose={this.resetError}
              closable
              message={error}
              showIcon
              type="error"
              style={{ marginTop: 24 }}
            />
          )}
        </Col>
      </Row>
    );
  }
}

/*LoginPage.propTypes = {
  onLogin: T.func.isRequired,
  isLogged: T.bool,
  history: T.shape({ replace: T.func.isRequired }).isRequired,
  location: T.shape({
    state: T.shape({ from: T.shape({ pathname: T.string }) }),
  }).isRequired,
};*/

export default LoginPage;



