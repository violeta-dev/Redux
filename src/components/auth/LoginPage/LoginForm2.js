import React from 'react';
import styles from './LoginForm.module.css';
import { Button, Checkbox, Input } from 'antd';
import { MailOutlined, LockOutlined } from '@ant-design/icons';

import LoginHOC from './LoginHOC';

const LoginForm = ({ email, password, remember, handleChange, handleSubmit,canSubmit }) => (

  <form onSubmit={handleSubmit}>
      <Input
        name="email"
        className={styles.input}
        prefix={<MailOutlined />}
        placeholder="Email"
        onChange={handleChange}
        value={email}
      />
      <Input.Password
        name="password"
        className={styles.input}
        prefix={<LockOutlined />}
        placeholder="Password"
        onChange={handleChange}
        value={password}
      />
      <Checkbox
        name="remember"
        className={styles.input}
        onChange={handleChange}
        checked={remember}
      >
        Remember me
      </Checkbox>
      <Button type="primary" htmlType="submit" disabled={!canSubmit()} block>
        Log In
      </Button>
    </form>
);

export default (LoginHOC(LoginForm));