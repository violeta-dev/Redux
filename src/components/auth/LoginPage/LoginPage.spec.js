import React from 'react';
import { shallow } from 'enzyme';

import { LoginPage } from './LoginPage';

describe('LoginPage', () => {
  const props = {
    loading: false,
    error: null,
    onLogin: jest.fn(),
  };

  const render = () => shallow(<LoginPage {...props} />);
  test('snapshot', () => {
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
  });
  
  test('should call a onLogin', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    const wrapper = render();
    const LoginForm = wrapper.find('LoginForm');
    LoginForm.hostNodes().simulate('submit', event);
    expect(event.preventDefault).toHaveBeenCalled();
    expect(props.onLogin).toHaveBeenCalled();
  });

});



