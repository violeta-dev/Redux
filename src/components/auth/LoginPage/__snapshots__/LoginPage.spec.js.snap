// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage snapshot 1`] = `ShallowWrapper {}`;
